multiaccounts/accounts/database_test.go

```diff
@@ -42,6 +42,27 @@ func TestGetAddresses(t *testing.T) {
 	require.Equal(t, []types.Address{{0x01}, {0x02}}, addresses)
 }
 
+func TestUpdateAccountPosition(t *testing.T) {
+	db, stop := setupTestDB(t)
+	defer stop()
+	accounts := []*Account{
+		{Address: types.Address{0x01}, Chat: true, Wallet: true},
+		{Address: types.Address{0x02}},
+	}
+	require.NoError(t, db.SaveOrUpdateAccounts(accounts))
+	accounts, err := db.GetAccounts()
+	require.NoError(t, err)
+	require.Equal(t, accounts[0].Position, int64(1))
+	require.Equal(t, accounts[1].Position, int64(2))
+
+	err = db.UpdateAccountPosition(accounts[1].Address, 1)
+	require.NoError(t, err)
+
+	accounts, err = db.GetAccounts()
+	require.Equal(t, accounts[0].Position, int64(0))
+	require.Equal(t, accounts[1].Position, int64(1))
+}
```

Comment:
The following extended tests fails a normal presentation layer usage

```suggestion
func TestUpdateAccountPosition(t *testing.T) {
	db, stop := setupTestDB(t)
	defer stop()
	accounts := []*Account{
		{Address: types.Address{0x01}, Chat: true, Wallet: true},
		{Address: types.Address{0x02}},
		{Address: types.Address{0x03}},
		{Address: types.Address{0x04}},
		{Address: types.Address{0x05}},
	}
	require.NoError(t, db.SaveOrUpdateAccounts(accounts))
	res, err := db.GetAccounts()
	require.NoError(t, err)
	validateOrderForRes := func(accOrd []int) {
		require.Equal(t, len(accounts), len(accOrd))
		for i, ord := range accOrd {
			var resJson []byte
			if accounts[ord].Address != res[i].Address {
				resJson, _ = json.Marshal(res)
			}
			require.Equal(t, accounts[ord].Address, res[i].Address, "account order broken at element %d; result content", i, string(resJson))
		}
	}
	validateOrderForRes([]int{0, 1, 2, 3, 4})

	err = db.UpdateAccountPosition(accounts[0].Address, 4)
	require.NoError(t, err)

	res, err = db.GetAccounts()
	require.NoError(t, err)
	validateOrderForRes([]int{1, 2, 3, 0, 4})

	err = db.UpdateAccountPosition(accounts[2].Address, 4)
	require.NoError(t, err)

	res, err = db.GetAccounts()
	require.NoError(t, err)
	validateOrderForRes([]int{1, 3, 0, 2, 4})
}
```

Reply:
I also fixed the tests to follow the way API should be used and they pass all right
```go
func TestUpdateAccountPosition(t *testing.T) {
	db, stop := setupTestDB(t)
	defer stop()
	accounts := []*Account{
		{Address: types.Address{0x01}, Chat: true, Wallet: true},
		{Address: types.Address{0x02}},
		{Address: types.Address{0x03}},
		{Address: types.Address{0x04}},
		{Address: types.Address{0x05}},
	}
	require.NoError(t, db.SaveOrUpdateAccounts(accounts))
	res, err := db.GetAccounts()
	require.NoError(t, err)
	validateOrderForRes := func(accOrd []int) {
		require.Equal(t, len(accounts), len(accOrd))

		sort.Slice(res, func(l, r int) bool {
			return res[l].Position < res[r].Position
		})

		for i, ord := range accOrd {
			var resJson []byte
			if accounts[ord].Address != res[i].Address {
				resJson, _ = json.Marshal(res)
			}
			require.Equal(t, accounts[ord].Address, res[i].Address, "account order broken at element %d; result content", i, string(resJson))
		}
	}
	validateOrderForRes([]int{0, 1, 2, 3, 4})

	err = db.UpdateAccountPosition(accounts[0].Address, res[3].Position)
	require.NoError(t, err)

	res, err = db.GetAccounts()
	require.NoError(t, err)
	validateOrderForRes([]int{1, 2, 3, 0, 4})

	err = db.UpdateAccountPosition(accounts[2].Address, res[3].Position)
	require.NoError(t, err)

	res, err = db.GetAccounts()
	require.NoError(t, err)
	validateOrderForRes([]int{1, 3, 0, 2, 4})
}
```